fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/Quantifier.java

```diff
@@ -211,11 +219,11 @@ public boolean equals(final Object o) {
         if (this == o) {
             return true;
         }
-        if (!(o instanceof ForEach)) {
+        if (!(o instanceof Quantifier)) {
             return false;
         }
-        final ForEach forEach = (ForEach)o;
-        return Objects.equals(getRangesOver(), forEach.getRangesOver());
+        final Quantifier quantifier = (Quantifier)o;
+        return Objects.equals(getRangesOver(), quantifier.getRangesOver());
```

Comment:
For a quantifier to be equal, doesn't the flavor also need to match? Is that used anywhere?

Reply:
it does -- fixing